rules/otbn.bzl

```diff
@@ -148,6 +148,45 @@ def _otbn_binary(ctx):
         ),
     ]
 
+def _otbn_test(ctx):
+    """This rule is for standalone OTBN unit tests, which are run on the host
+    via the OTBN simulator.
+
+    It first generates binaries using the same method as otbn_binary, then runs
+    them on the simulator. Tests are expected to count failures in the w0
+    register; the test checks that w0=0 to determine if the test passed.
+    """
+    providers = _otbn_binary(ctx)
+
+    # Extract the output .elf file from the output group.
+    elf = providers[1].elf.to_list()[0]
+
+    # Create a simple script that runs the OTBN simulator on the .elf file and
+    # checks if the w0 register is 0.
```

Comment:
Not necessarily for this PR, but I wonder whether it would make sense to merge the "simple test" stuff that's currently in `hw/ip/otbn/dv/otbnsim/test` with this? That has a notion of an expected result: maybe the two could use the same logic?

Reply:
I didn't know about that file, thanks for pointing it out! Since the logic there already checks for all registers having the expected value, that will be easier than porting existing test to the `w0=0` convention. It looks like it'll require some nontrivial adjustment of the script, though, so I made #12223 to track this.